### ISSUE TITLE
Issues after installation

### ISSUE THREAD

**Issue by Payamch** [husisusi/officeonlin-install.sh#67]
Hello,
So I tried following steps in order to instal LibreOffice on Ubuntu 16.04.2 LTS

1.  run the script without any changes to the remote link
2. run the scrip with changing the remote links as discussed [here](https://github.com/husisusi/officeonlin-install.sh/issues/49) 

I have difficulties and some bugs after the installation after both methods. The bugs that I have are some UI bugs (rows get mixed up or columns get mixed up) as in the screenshot

![image](https://user-images.githubusercontent.com/16401566/26878472-e04c75dc-4b8d-11e7-8030-407e51763a06.png)

I also tried to delete the opt folder and reinstall collabora with different SHA versions, because if you open a collabora demo account the commit will be on 52f0568e so I also tried to install it on that commit but still no luck.

![image](https://user-images.githubusercontent.com/16401566/26878573-45cb66e8-4b8e-11e7-9f78-413826af88da.png)

Any suggestions what should I do?

**Comment by Kassiematis** [comment]
I also have a lot of problems in Libreoffice Online. Especially with column and row resizing. Spreadsheets with filters, formulas etc. can also be tricky.
I think that the demo version is different from the open source one. Not al code is supplied with the open source version.

**Comment by togglecode** [comment]
Yes you are right, the demo version seems to be based on collabora which seems [not to be opensource](https://www.collaboraoffice.com/community-en/understanding-the-differences-between-libreoffice-online-code-and-collabora-online/). Well, I thought only I have those issues, thank you.

**Comment by Kassiematis** [comment]
The Collabora demo seems to get their source for Libreoffice from https://github.com/LibreOffice/core.git and not http://download.documentfoundation.org/libreoffice. Could that be the difference?

**Comment by aalaesar** [comment]
that's an interesting point
the fact that latest commit in LoOnline master branch could be based on LibreOffice master branch instead of a release.
It would be interesting to check how Loolwsd is more functional when compiled  from a branch named after a LO release instead of master, like libreoffice-5-4 for example, if one's using libreoffice 5.4.x behind of course...

**Comment by Kassiematis** [comment]
I am compiling as we speak. :-)

**Comment by aalaesar** [comment]
did you specified the branch relative to the LO release ?

**Comment by Kassiematis** [comment]
Actually not, the demo is using version cp-5.3-13 but I am using the default.

**Comment by Kassiematis** [comment]
I have restarted the compile with the following clone.
git clone -b "distro/collabora/cp-5.3" "https://github.com/LibreOffice/core.git" "core"

**Comment by togglecode** [comment]
@kassiematis So are the problems still there even after you fetched the project from core?

**Comment by Kassiematis** [comment]
The compile is still running.

**Comment by Kassiematis** [comment]
It is definitely an improvement with the different github branch. I can resize columns.
Row 1 and column A show their number and letter again. (that's how far I have tested)
Question is how to change the script to incorporate the clone command. 
`git clone -b "distro/collabora/cp-5.3" "https://github.com/LibreOffice/core.git" "/opt/libreoffice"`

**Comment by Kassiematis** [comment]
![1f67 tmp](https://user-images.githubusercontent.com/26070573/26900623-21387ec0-4bd3-11e7-87ff-fb65070e8e67.jpg)

**Comment by togglecode** [comment]
Thank you for the update, that's a great improvement!

So basically you cloned the new core in /opt/libreoffice folder and ran the officeonlin-install.sh script again? Or did you change some parameters in the script ?

**Comment by Kassiematis** [comment]
The script will not work as intended without some mayor changes.

I ran the commands

sudo -s
systemctl stop loolwsd.service
rm -r /opt/libreoffice
`git clone -b "distro/collabora/cp-5.3" "https://github.com/LibreOffice/core.git" "/opt/libreoffice"`
chown -R lool:lool /opt/libreoffice
cd /opt/libreoffice
sudo -Hu lool ./autogen.sh --without-help --without-myspell-dicts
sudo -Hu lool make

This is from the top of my head so beware of typos.

**Comment by husisusi** [comment]
@Kassiematis 
What about limitation (10 Docs/20 Connections)? 
Why you compiled it like that?
It is interesting to know if we can build it with no limitation.

**Comment by Kassiematis** [comment]
Only libreoffice gets compiled. Loolwsd is not changed at all so the limitation is still 100/200.

**Comment by husisusi** [comment]
You're right! @Kassiematis

**Comment by aalaesar** [comment]
So you put in evidence that both loolwsd and LibreOffice should be compiled from synchronized branch
aka using master /master for both software 👍 
so far the script uses libreoffice official releases (auto-find the latest release by default) and the git repo for loolwsd with master branch by default. and so we have found issue with this.
before doing some rework (which is not that big) just after Only one test (!!) we may need to test the other way:
use the official libreoffice release and use the loolwsd branch related to the release.
for example:
actual libreoffice latest release is 5.3.3 so test with setting` lool_src_branch` to `libreoffice-5-3` (latest commit 16 dec sic) and see if the problems are also solved.

my though is : if we switch to building libreoffice from master branch ,we're gonna have a bad time with temp broken builds and constantly changing dependencies. It will be a nightmare to maintain and support users.
I was not happy with using latest commit from master for loowsd here, But without proper release, it was hard to choose how to select best commits. We may have the beginning of an answer here: choose a branch set for the libreoffice correct version.

**Comment by Kassiematis** [comment]
I can confirm that the collabora branch is the way to go. It is maintained by the same people that develops for loolwsd. The differences between the master and collabora branch are significant and the master branch cannot be used in combination with loolwsd.

**Comment by aalaesar** [comment]
I've tested branch libreoffice-5-3 but something seems broken in the loleafleat as it missing lots of js files.
so by default the test conclude we can't trust "synchronised branch" for getting a stable product with the current version of the script.

@Kassiematis can you elaborate about the collabora branch ? for LO and/or  Lool  ?

**Comment by Kassiematis** [comment]
Lool = https://github.com/LibreOffice/online.git
LO = https://github.com/LibreOffice/core/tree/distro/collabora/cp-5.3

**Comment by aalaesar** [comment]
Ok I'm in.
I'll work this change

**Comment by Kassiematis** [comment]
The Collabora demo is using this branch for loolwsd. I will test this tommorow if that makes a difference with the master. 
https://github.com/LibreOffice/online/tree/distro/collabora/collabora-online-2-1
This branch seems to follow the LO branch from Collabora. You can see comments like this with the commits.

`Requires: Collabora Office 5.3-13`

**Comment by Kassiematis** [comment]
I have compiled with the following branches.
loolwsd = distro/collabora/collabora-online-2-1
libreoffice = distro/collabora/cp-5.3

Which are the same as the Collabora demo uses. Everything works fine. 

![collabora](https://user-images.githubusercontent.com/26070573/26952787-358da93e-4ca7-11e7-8e08-c279d9ebff66.png)

**Comment by Kassiematis** [comment]
Compiling with the following branches together also works fine.

loolwsd = https://github.com/LibreOffice/online/tree/libreoffice-5-4
libreoffice = https://github.com/LibreOffice/core/tree/libreoffice-5-4

only this directory has to be created and owner altered to lool
 /usr/local/var/cache/libreoffice-online

**Comment by Gomez** [comment]
This is valuable information. Maybe its a good idea to pin a "kown/good working set" in the script. I already do this, as documented here:

https://github.com/husisusi/officeonlin-install.sh#versions

Would be good to make this "kown/good working set" the default and update versions with PRs if its approved?

**Comment by aalaesar** [comment]
Nice 👍 
> only this directory has to be created and owner altered to lool
/usr/local/var/cache/libreoffice-online

had the same with libreoffice-5-3

It's a shame the libreoffice-5-3 branch of online seems broken (in my tests at least), The branch seems abandonned : last commit was last year also. Could appen again.
I think we should keep on collabora branches _by default_ as you suggested

**Comment by enoch85** [comment]
Just wanted to inform you guys...

I just compiled from the script. It took 5 hours on a 4 core CPU i5 6500 series. I didn't change anything, I just ran it - and this is the final result:

```
Tested file:///opt/libreoffice/vcl/qa/cppunit/graphicfilter/data/bmp/fail/EDB-24743-1.bmp: Fail (0ms)
Testing file:///opt/libreoffice/vcl/qa/cppunit/graphicfilter/data/bmp/fail/EDB-24743-4.bmp:
Tested file:///opt/libreoffice/vcl/qa/cppunit/graphicfilter/data/bmp/fail/EDB-24743-4.bmp: Fail (0ms)
Testing file:///opt/libreoffice/vcl/qa/cppunit/graphicfilter/data/bmp/fail/crash-1.bmp:
Tested file:///opt/libreoffice/vcl/qa/cppunit/graphicfilter/data/bmp/fail/crash-1.bmp: Fail (1ms)
Testing file:///opt/libreoffice/vcl/qa/cppunit/graphicfilter/data/bmp/fail/nodict-compress.bmp:
Killed


No core file identified in directory /opt/libreoffice/workdir/CppunitTest/vcl_filters_test.test.core
To show backtraces for crashes during test execution,
enable core files with:

   ulimit -c unlimited


Error: a unit test failed, please do one of:
make CppunitTest_vcl_filters_test CPPUNITTRACE="gdb --args"
    # for interactive debugging on Linux
make CppunitTest_vcl_filters_test VALGRIND=memcheck
    # for memory checking
make CppunitTest_vcl_filters_test DEBUGCPPUNIT=TRUE
    # for exception catching

make[1]: *** [/opt/libreoffice/workdir/CppunitTest/vcl_filters_test.test] Error 1
/opt/libreoffice/solenv/gbuild/CppunitTest.mk:101: recipe for target '/opt/libreoffice/workdir/CppunitTest/vcl_filters_test.test' failed
make[1]: *** Waiting for unfinished jobs....
Makefile:263: recipe for target 'build' failed
make: *** [build] Error 2
```

In other words, script is failing in current state.

**Comment by Kassiematis** [comment]
Looks like the problem is with the Libreoffice compile. Found this on internet.
https://forums.freebsd.org/threads/60837/
Aalaesar is changing the script to use a different branch for LO. That will probably solve the problem.

**Comment by enoch85** [comment]
@Kassiematis Ok thanks, trying to compile again just to be sure. This time with 12 cores so it's going a bit faster. :) Will let you know.

**Comment by aalaesar** [comment]
@Kassiematis @Payamch @husisusi 
> Aalaesar is changing the script to use a different branch for LO. That will probably solve the problem.>

Working branch here:
https://github.com/aalaesar/officeonlin-install.sh/tree/create_set
The "set" new feature is working on it's own and I'm testing inside the script at the moment.
it's still a little rough thought . I'll refine the changes once the script is more solid.
but further testing would be welcome 👍

**Comment by Kassiematis** [comment]
@aalaesar 

First test worked perfect.

LibreOffice Online
Deze versie van LibreOffice Online werk op basis van

LOOLWSD
2.1.2 (git hash: 51196ee)

LOKit
Collabora OfficeDev 5.3-16 (git hash: d134b1f)

**Comment by aalaesar** [comment]
Nice 👍
What's your setup ?
What's your commit hash on the working branch ? (current commit is [d022039](https://github.com/aalaesar/officeonlin-install.sh/commit/d022039a3b7f9c9b8e7ea40502b23c395e020fa1))
Because I have refined some functions and now the script compile by default from the latest **tag** of the collabora set . ([fd4bc0e](https://github.com/aalaesar/officeonlin-install.sh/commit/fd4bc0ea80a7b1a2c6924cd96ec16141e57e25f8))
But the hashes on your info pages are not the latest tags
_LibreOffice core:_
![image](https://user-images.githubusercontent.com/18649249/27042846-7094ab9c-4f98-11e7-90a0-df9923fb5a02.png)

_LibreOffice Online:_
![image](https://user-images.githubusercontent.com/18649249/27042781-4a162bc6-4f98-11e7-99be-615cb8df1983.png)

So maybe you changed the configuration or you kept on [6d38b63](https://github.com/aalaesar/officeonlin-install.sh/commit/6d38b63de5443f66fbdf174c82bd175d1b87d803).
What do you think ? is the "build from latest tag" feature broken ?

**Comment by Kassiematis** [comment]
I dont know which version of the script i used, I downloaded the latest this morning. Have to check tomorrow.

**Comment by Kassiematis** [comment]
Just started new compile with the latest script. 

Searching for a set named collabora...
core branch: distro/collabora/cp-5.3
core branch: distro/collabora/collabora-online-2-1
...
HEAD is now at e19a986 tdf#101885: fix crash in AccFrameSelector

**Comment by Kassiematis** [comment]
@aalaesar 

After compilation

LibreOffice Online
Deze versie van LibreOffice Online werk op basis van

LOOLWSD
2.1.2 (git hash: 4b03b9b)

LOKit
Collabora OfficeDev 5.3-16 (git hash: e19a986)

**Comment by aalaesar** [comment]
👍
Work fine by default👌
 I'll try another set.

**Comment by Kassiematis** [comment]
I get an error because of this command in the script.

`if pgrep -u lool loolwsd; then`

No command 'pgrep:' found, did you mean:
 Command 'pgrep' from package 'procps' (main)
pgrep:: command not found

**Comment by aalaesar** [comment]
Thanks for the report.
This error append during the final test
what system do you use ?
I found pgrep pretty standard on Ubuntu 14.04, 16.04 & debian 8 so I didn't bother adding the package 😕

**Comment by Kassiematis** [comment]
I am using Ubuntu 16.04

**Comment by aalaesar** [comment]
OK
well I suppose I'll add procps in the package list
so you may have the basic/test loolwsd instance running on your host now
old command to find loolwsd and kill it is `ps -u lool | grep loolwsd` 
and `ps -u lool -o pid,cmd | grep loolwsd |awk '{print $1}' | xargs kill`

**Comment by aalaesar** [comment]
closed by #69  ?

**Comment by enoch85** [comment]
Just want to confirm, my test from 5 days ago was successful. Great work guys. :)
